### DISCUSSION TITLE
Redirect with useRoutes

### DISCUSSION THREAD

**Discussion by outbackStack** [solidjs/solid-router#151] [Q&A] [answered]
Thank you for this amazing router. I'm in the process of moving over from React.

Question: How could I redirect to a page with useRoutes?

With react-router I could this. If you go to path "/" then it will redirect you to "/dashboard"

```js
const routes = [
    {
      path: "/",
      element: <Layout />,
      children: [
          { index: true, element: <Navigate to="/dashboard" /> },
          {
            path: "other",
            element: <OtherPage />,
          }
      ]
    }
]
```

**Comment by ryansolid** [comment, marked as answer]
This won't work the way you thing it will because Solid doesn't use VDOM and the way compilation works. If you want to use `useRoutes` typically don't use JSX. And use `component` instead of `element`.

However to get this to work simulating our JSX compilation I'd try:

```js
const routes = [
    {
      path: "/",
      get element() { return <Layout /> },
      children: [
          { path: "/", get element() { return <Navigate to="/dashboard" /> } },
          {
            path: "other",
            get element() { return <OtherPage /> },
          }
      ]
    }
]
```

**Reply by outbackStack** [reply]
Thank you @ryansolid !

This works! :-)
